flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphRestartTest.java

```diff
@@ -581,6 +565,106 @@ public void testSuspendWhileRestarting() throws Exception {
 		assertEquals(JobStatus.SUSPENDED, eg.getState());
 	}
 
+	@Test
+	public void testConcurrentLocalFailAndRestart() throws Exception {
```

Comment:
Right, this one was a test that should have been there in the first place and I took this chance to add it.